deepchem/models/lightning/dc_lightning_module.py

```diff
@@ -28,15 +75,40 @@ def __init__(self, dc_model):
             TorchModel to be wrapped inside the lightning module.
         """
         super().__init__()
+        self.save_hyperparameters(ignore=['dc_model'])
         self.dc_model = dc_model
-
-        self.pt_model = self.dc_model.model
-        # This will not work for ModularTorchModel as it is directly uses `loss_func` to compute loss.
-        self.loss = self.dc_model._loss_fn
```

Comment:
is the self.loss not needed here anymore? please explain how back compatibility is maintained in this DCLightningModule class. Why did you modify the existing class instead of making a new class or a child class. What's the need here?